themes/bootstrap3/templates/RecordTab/staffviewarray.phtml

```diff
@@ -8,9 +8,13 @@
     <tr>
       <th><?=$this->escapeHtml($field)?></th>
       <td>
-        <?php foreach ((array)$values as $value): ?>
-          <?=$this->escapeHtml(is_array($value) ? print_r($value, true) : $value)?><br />
-        <?php endforeach; ?>
+        <?php
+          $values = (array)$values;
+          if (count($values) == 1 && is_int(array_key_first($values))) {
```

Comment:
What is the purpose of this special case? It seems to me that if $values is ever a non-array value, it will get cast to an array and then shifted back into a non-array. Is this intentional?

Reply:
I wasn't sure why the cast to array was there before, so I left it in assuming it had a purpose.

Previously, all keys for arrays at the top level were not being displayed, whether integer or string. The above still removes the extra indentation level if there is only a single integer key (avoiding the extra indentation for the whole block of array data, like before), but will now show string key values in the output. Also, for arrays with multiple integer keys, it will add the hyphen to show when the next array value begins.

I made an assumption was that top-level keys being hidden previously was to avoid the common case of having single integer top-level key and wanting to skip displaying it. The above code was trying to preserve that.

Reply:
The previous (array) cast was useful because it allowed us to iterate in a foreach loop over a value whether or not it was an array. Since the loop has been eliminated, I think you could probably now safely remove the cast, and instead just add an `is_array($values)` condition to the beginning of your if statement. That might make the code slightly more straightforward by eliminating the case where we create and then immediately unpack an array.

Reply:
Thanks for explaining, Demian. That makes a lot of sense. I've updated the branch; let me know if there is anything else that could be improved.